R/bmerge.R

```diff
@@ -84,7 +84,18 @@ bmerge = function(i, x, icols, xcols, roll, rollends, nomatch, mult, ops, verbos
           next
         }
       }
-      stopf("Incompatible join types: %s (%s) and %s (%s). Factor columns must join to factor or character columns.", xname, x_merge_type, iname, i_merge_type)
+      condition_message = sprintf("Incompatible join types: %s (%s) and %s (%s). Factor columns must join to factor or character columns.", xname, x_merge_type, iname, i_merge_type)
+      condition <- structure(
+        list(
+          message = condition_message,
+          bmerge_x_arg_col_name      = names(x)[xcol],
+          bmerge_x_arg_type          = x_merge_type,
+          bmerge_i_arg_col_name      = names(i)[icol],
+          bmerge_i_arg_type          = i_merge_type
+        ),
+        class = c("bmerge_incompatible_type_error", "data.table_error", "error", "condition")
+      )
+      stop(condition)
```

Comment:
`stopf()` adds a `call` entry to the error object, too:

https://github.com/Rdatatable/data.table/blob/5bbc4d516eb9a05471e7b5221c8c23a540af97a4/R/translation.R#L8

That will let this new error object play well with `conditionCall()`